File: spring_java_orm_sample/src/main/java/com/example/demo/util/ExceptionHandlerUtil.java
package com.example.demo.util;

import jakarta.validation.Path;
import org.jspecify.annotations.Nullable;

import java.util.Map;
import java.util.Objects;

public final class ExceptionHandlerUtil {
    public static Map<String, String> createValidationError(@Nullable String field, @Nullable String message) {
        return Map.of(
            "field", Objects.toString(field, ""),
            "message", Objects.toString(message, "")
        );
    }

    public static String getLastPropertyName(Path propertyPath) {
        String propertyName = "";
        for (final var node : propertyPath) {
            if (node.getName() != null) {
                propertyName = node.getName();
            }
        }
        return propertyName;
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/util/ExceptionHandlerUtil.kt
package com.example.demo.util

import jakarta.validation.Path
import java.util.*

object ExceptionHandlerUtil {
    fun createValidationError(field: String?, message: String?): Map<String, String> {
        return mapOf(
            "field" to Objects.toString(field, ""),
            "message" to Objects.toString(message, "")
        )
    }

    fun getLastPropertyName(propertyPath: Path): String? {
        var propertyName: String? = ""
        for (node in propertyPath) {
            if (node.getName() != null) {
                propertyName = node.getName()
            }
        }
        return propertyName
    }
}
